frontend/src/metabase/query_builder/components/filters/modals/InlineValuePicker/InlineValuePicker.tsx

```diff
@@ -45,16 +60,39 @@ export function InlineValuePicker({
         operators={filterOperators}
         onOperatorChange={changeOperator}
       />
-      <ArgumentSelector
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore: this component doesn't have types or propTypes
-        value={filter.arguments()}
-        onChange={changeArguments}
-        className="input"
-        fields={[field]}
-        showOptionsInPopover
-        multi
-      />
+      {!hideArgumentSelector && !isBetween && (
```

Comment:
nit: would extracting these 2 blocks of JSX into a single component would make it more obvious that there are only 3 results when rendering the argument input
- No argument input
- Single argument
- 2 arguments (only `is-between`)

Reply:
good call! i factored this out and it's much cleaner now.